web/src/beta/lib/core/engines/Cesium/pickMany.ts

```diff
@@ -0,0 +1,302 @@
+// Picks multiple objects in the given rectangle.
+function pickFramebufferEnd(
+  this: PickFramebuffer,
+  screenSpaceRectangle: BoundingRectangle,
+): object[] {
+  const width = screenSpaceRectangle.width ?? 1;
+  const height = screenSpaceRectangle.height ?? 1;
+  const context = this._context;
+  const pixels = context.readPixels({
+    x: screenSpaceRectangle.x,
+    y: screenSpaceRectangle.y,
+    width,
+    height,
+    framebuffer: this._fb.framebuffer,
+  });
+
+  // TODO: Why this assertion fails, even if the format is RGBA and data type
+  // is unsigned bytes?
+  // invariant(pixels.length % 4 === 0)
+
+  // I don't know why Cesium doesn't provide this functionality but reads pixels
+  // in spiral to find only a single object. Precision issue maybe?
+  const objects = new Set<object>();
+  for (let index = 0; index < pixels.length; index += 4) {
+    colorScratch.red = Color.byteToFloat(pixels[index]);
+    colorScratch.green = Color.byteToFloat(pixels[index + 1]);
+    colorScratch.blue = Color.byteToFloat(pixels[index + 2]);
+    colorScratch.alpha = Color.byteToFloat(pixels[index + 3]);
+    const object = context.getObjectByPickColor(colorScratch);
+    if (object != null) {
+      objects.add(object);
+    }
+  }
+  return Array.from(objects.values());
+}
+
+const pickingFrustumScratch = new PerspectiveOffCenterFrustum();
+const pixelSizeScratch = new Cartesian2();
+
+// Derived from: https://github.com/CesiumGS/cesium/blob/1.105/packages/engine/Source/Scene/Picking.js#L197
+// Changes are in the comments below.
+function getPickCullingVolume(
+  scene: Scene,
+  drawingBufferPosition: Cartesian2,
+  width: number,
+  height: number,
+  viewport: BoundingRectangle,
+): CullingVolume {
+  const camera = scene.camera;
+  const frustum = camera.frustum;
+  invariant(frustum instanceof PerspectiveFrustum);
+
+  const near = frustum.near;
+  const tanPhi = Math.tan(frustum.fovy / 2);
+  const tanTheta = frustum.aspectRatio * tanPhi;
+
+  const bx = drawingBufferPosition.x;
+  const by = drawingBufferPosition.y;
+  const x = (2 * (bx - viewport.x)) / viewport.width - 1;
+  const y = (2 * (viewport.height - by - viewport.y)) / viewport.height - 1;
+  const xDir = x * near * tanTheta;
+  const yDir = y * near * tanPhi;
+
+  const pixelSize = frustum.getPixelDimensions(
+    viewport.width,
+    viewport.height,
+    near, // Changed from 1
+    getPixelRatio(scene), // Changed from 1
+    pixelSizeScratch,
+  );
+  const pickWidth = (pixelSize.x * width) / 2;
+  const pickHeight = (pixelSize.y * height) / 2;
+
+  const offCenter = pickingFrustumScratch;
+  offCenter.top = yDir + pickHeight;
+  offCenter.bottom = yDir - pickHeight;
+  offCenter.right = xDir + pickWidth;
+  offCenter.left = xDir - pickWidth;
+  offCenter.near = near;
+  offCenter.far = frustum.far;
+
+  return offCenter.computeCullingVolume(camera.positionWC, camera.directionWC, camera.upWC);
+}
+
+function getIntersection(
```

Comment:
Its better to minimize in functional programming. here It's modifying result directly. Instead, consider returning a new object:

function getIntersection(
  a: BoundingRectangle,
  b: BoundingRectangle,
  result = new BoundingRectangle(),
): BoundingRectangle | undefined {
  //...
  return {
    x: x1,
    y: y1,
    width: x2 - x1,
    height: y2 - y1,
  };
  
  Just as a practice if you can apply similar thing to other functions that'll be great 🙏